azerothcore/azerothcore-wotlk
# fix(Scripts/Scholomance) Grandmaster Gandling correctly drops threat if teleports the tank

## Changes Proposed:
-  Rewrote the whole thing. Threw away all the SAI

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1806 (In practice not, he can still go through locked doors, but that's only when he has no choice)

## SOURCE:


## Tests Performed:

- builds and run on my CI. Tested with 2 clients and it seems the reset works, summoning mobs works, doors opening works, mobs running out on death works.


## How to Test the Changes:


1. Whole event has been rewritten, therefore everything needs tested
2. What happens if a player dies within a locked room after summon (mobs should run out)
3. Does he correctly take a new target after casting portal if he targets the main tank?
4. Are spells correctly cast?
5. Does he reset correctly?
6. Yells?
7. Do mobs always spawn in the right rooms?
8. Does he send multiple people to the same room?

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
